.github/workflows/yocto-build-deploy.yml

```diff
@@ -121,37 +116,47 @@ on:
         required: false
         type: boolean
         default: ${{ startsWith(github.event.push.ref, 'refs/tags/v20') }} # Set this flag if the tag looks like an esr tag - this is good for 100 years or until balenaOS v20 comes out
+      # FIXME: add to GitHub environments
       aws_iam_role:
         description: "IAM role to assume for S3 permissions"
         required: false
         type: string
-        default: "arn:aws:iam::567579488761:role/balena-os-deploy-s3-access-Role-8r0SXqfzgolr" #TODO - Replace using environments
+        default: "arn:aws:iam::567579488761:role/balena-os-deploy-s3-access-Role-8r0SXqfzgolr"
+      # FIXME: add to GitHub environments or use S3_REGION
       aws_region:
         description: "AWS region"
         required: false
         type: string
         default: "us-east-1"
-      test-suites:
-        description: "Comma-separated list of test suites to run. No spaces in between."
-        required: false
-        type: string
-        default: "os,hup,cloud"
-      test-workers:
-        description: "Comma-separated list of worker types to use for testing. No spaces in between. The valid worker types are `qemu` and `testbot`."
-        required: false
-        type: string
-        default: "testbot" # Most devices use "testbot" , so qemu is the exception and should be set at the device repo level
-      worker-fleets:
-        description: "Testbot fleet for finding available Leviathan workers. Not used for QEMU workers. Can accept a list of apps separated by commas, no spaces in between."
-        type: string
+      # Supported fields for the test matrix:
+      # - test_suite: (required) The test suite to run. The valid test suites are `os`, `hup`, and `cloud`.
+      # - environment: (required) The balenaCloud environment to use for testing, e.g. `bm.balena-dev.com` or `balena-cloud.com.
```

Comment:
Remove the trailing fullstops in these sentences to reduce a little bit of the confusion.